docs/metadata-extract-embbed.md

```diff
@@ -0,0 +1,303 @@
+        extractor.extractMetadata(sourceMimetype, transformOptions, sourceFile, targetFile);
+    }
+
+    // Similar code for embedMetadata
+~~~
+
+### AbstractMetadataExtractor
+The `AbstractMetadataExtractor` may be extended to perform metadata extract and embed tasks, by overriding two methods
+in the sub classes:
+~~~
+    public abstract Map<String, Serializable> extractMetadata(String sourceMimetype, Map<String, String> transformOptions,
+                                                                  File sourceFile) throws Exception;
+
+    public void embedMetadata(String sourceMimetype, String targetMimetype, Map<String, String> transformOptions,
+                              File sourceFile, File targetFile) throws Exception
+    {
+        // Default nothing, as embedding is not supported in most cases
+    }
+~~~
+Method parameters:
+
+* **sourceMimetype** mimetype of the source
+* **transformOptions** transform options from the client
+* **sourceFile** the source as a file
+
+* The `extractMetadata` should extract and return ALL available metadata from the sourceFile.
+* These values are then mapped into content repository property names and values, depending on what is defined in a
+  `<classname>_metadata_extract.properties`"}` file. Value may be discarded or a single value may even be used for
+  multiple properties.
+* The selected values are set back to the content repository as JSON as a mapping of fully qualified content repository
+  property names to values, where the values are applied to the source node.
+
+### &lt;classname>_metadata_extract.properties
+
+The `AbstractMetadataExtractor` class reads the `<classname>_metadata_extract.properties` file, so that it knows how to
+map metadata returned from the sub class `extractMetadata` method onto crontent repository properties. The following is
+an example for an email (file extension eml):
+
+~~~
+#
+# RFC822MetadataExtractor - default mapping
+#
+
+# Namespaces
+namespace.prefix.imap=http://www.alfresco.org/model/imap/1.0
+namespace.prefix.cm=http://www.alfresco.org/model/content/1.0
+
+# Mappings
+messageFrom=imap:messageFrom, cm:originator
+messageTo=imap:messageTo, cm:addressee
+messageCc=imap:messageCc, cm:addressees
+messageSubject=imap:messageSubject, cm:title, cm:description, cm:subjectline
+messageSent=imap:dateSent, cm:sentdate
+messageReceived=imap:dateReceived
+Thread-Index=imap:threadIndex
+Message-ID=imap:messageId
+~~~
+As can be seen, the email's metadata for `messageFrom` (if available) will be used to set two properties in the content
+repository (if they exist): `imap:messageFrom`, `cm:originator`. The property names use namespace prefixes specified above.
+
+
+### overwritePolicy
+It is possible to specify if properties in the content repository will be set if the extracted values are not null or if
+the properties are already have a value. By default, `PRAGMATIC` is used. Generally you will not need to change this.
```

Comment:
extra 'are' in  'the properties are already have a value'

Reply:
thanks